File: solutions/c/scrabble-score/1/scrabble_score.c
#include "scrabble_score.h"
#include <string.h>

unsigned int score_letter(char ltr) {
    if (ltr < 97) ltr+= 32;
    if (ltr > 122) return 0;
    switch (ltr) 
    {
        case 'z' : return 10;
        case 'q' : return 10;
        case 'x' : return 8;
        case 'j' : return 8;
        case 'k' : return 5;
        case 'y' : return 4;
        case 'w' : return 4;
        case 'v' : return 4;
        case 'h' : return 4;
        case 'f' : return 4;
        case 'p' : return 3;
        case 'm' : return 3;
        case 'c' : return 3;
        case 'b' : return 3;
        case 'd' : return 2;
        case 'g' : return 2;   
        default: return 1;        
    }    
}

unsigned int score(const char *word){
        int length = strlen(word);
        int sum = 0; 
        if( 0 < length)
         {
            for (int i=0; i<length; i++) 
             {
                    sum += score_letter(word[i]); 
             }
          }
        return sum;       
}

File: solutions/julia/scrabble-score/1/scrabble-score.jl
scrabble_dict = 
    Dict(1=>['A', 'E', 'I', 'O', 'U', 'L', 'N', 'R', 'S', 'T' ],
         2=>['D', 'G'],
         3=>['B', 'C', 'M', 'P' ],
         4=> ['F', 'H', 'V', 'W', 'Y'],  
         5=> ['K'], 
         8=> ['J','X'],
         10=> ['Q','Z'])

score_letter(lt) =
begin
    for score in keys(scrabble_dict) 
        if lt in scrabble_dict[score]
            return score
        end
    end
    return 0    
end

function score(str)
    score = 0
     for l in str
         score+=score_letter(uppercase(l))
     end
    score
end
